src/download.test.js

```diff
@@ -1,28 +1,33 @@
 const axios = require("axios");
-const { DownloadFile } = require("./download");
+const tmp = require("tmp");
+
+const { downloadFile } = require("./download");
 
 jest.mock("axios");
 
 const portalUrl = "https://siasky.net";
 const skylink = "XABvi7JtJbQSMAcDwnUnmp2FKDPjg8_tTTFP4BwMSxVdEg";
+const tmpFile = tmp.fileSync();
 
 describe("download", () => {
-  const dst_file = "./dst.txt";
+  const dst_file = tmpFile.name;
   const body = "asdf";
 
   axios.get.mockResolvedValue({ data: { body, pipe: function () {} } });
 
   it("should send get request to default portal", () => {
-    DownloadFile(dst_file, skylink);
+    downloadFile(dst_file, skylink);
 
     expect(axios.get).toHaveBeenCalledWith(`${portalUrl}/${skylink}`, { responseType: "stream" });
   });
 
   it("should use custom options if defined", () => {
-    DownloadFile(dst_file, skylink, {
+    downloadFile(dst_file, skylink, {
       portalUrl: "localhost",
     });
 
     expect(axios.get).toHaveBeenCalledWith(`localhost/${skylink}`, { responseType: "stream" });
   });
 });
+
+tmpFile.removeCallback();
```

Comment:
You need to create new file for each test either in each test separately or in `beforeEach` block. Then you should run this callback respectively either after each test separately or in `afterEach` block. 

Tests are async and right now the tmp file is being created and removed even before the tests are run. It also creates just one file that would be reused in two tests. It will most likely cause some errors or could produce false positives.